rholang/src/test/scala/coop/rchain/rholang/interpreter/matcher/StreamTSpec.scala

```diff
@@ -0,0 +1,63 @@
+package coop.rchain.rholang.interpreter.matcher
+
+import cats.{Eq, Id}
+import cats.data.WriterT
+import cats.laws.discipline.{MonadTests, MonoidKTests}
+import cats.tests.CatsSuite
+import monix.eval.Coeval
+import org.scalacheck.{Arbitrary, Gen}
+import org.scalatest.{FlatSpec, Matchers}
+
+class StreamTSpec extends FlatSpec with Matchers {
+
+  behavior of "StreamT"
+
+  it must "not trigger spurious evaluation of underlying stream for a lazy monad" in {
+    val stream: Stream[Int] = Stream.cons(1, ???)
+    var wrapped: StreamT[Coeval, Int] = null
+
+    noException shouldBe thrownBy {
+      stream.head
+      wrapped = StreamT.fromStream[Coeval, Int](Coeval.now(stream))
+    }
+
+    val next = wrapped.next
```

Comment:
The code below could be extracted to a separate test case

Reply:
It's going to be changed once StreamT is made even more lazy according to the comment below - then it belongs to this test.